addons/mail/static/tests/thread/attachment_list_tests.js

```diff
@@ -171,6 +171,38 @@ QUnit.test("view attachment", async () => {
     await contains(".o-FileViewer");
 });
 
+QUnit.test("view link-type pdf attachment", async (assert) => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({
+        channel_type: "channel",
+        name: "channel1",
+    });
+    const attachmentId = pyEnv["ir.attachment"].create({
+        name: "url.pdf.example",
+        mimetype: "application/pdf",
+        type: "url",
+        url: "https://pdfobject.com/pdf/sample.pdf",
+    });
+    pyEnv["mail.message"].create({
+        attachment_ids: [attachmentId],
+        body: "<p>Test</p>",
+        model: "discuss.channel",
+        res_id: channelId,
+        message_type: "comment",
+    });
+    const { openDiscuss } = await start();
+    openDiscuss(channelId);
+    await click(".o-mail-AttachmentCard", { text: "url.pdf.example" });
+    await contains(".o-FileViewer");
+    const route = url(`/web/content/${attachmentId}`, { filename: "url.pdf.example" });
+    // Hard coded route in the viewer from `get defaultSource()` in file_model.js
+    assert.hasAttrValue(
+        $(".o-FileViewer-view"),
+        "data-src",
+        `/web/static/lib/pdfjs/web/viewer.html?file=${encodeURIComponent(route)}#pagemode=none`
+    );
```

Comment:
Comment covers also the `const route`, so should be above.

This is also not a good idea to refer to implementation details in test: the test should try to stay as functionally significant as possible regardless of implementation detail. Here you want to see the pdf is viewable inside odoo pdf viewer, so checking (part of) url with `web/viewer.html` is good. Comment can tell this is route for odoo pdf viewer, no need to tell it's function X in file Y, they become very likely a lie whenever code changes, and comments are very poor to maintain due to their nature.